### DISCUSSION TITLE
about invalid syntax on code

### DISCUSSION THREAD

**Discussion by AhmetGG** [pyg-team/pytorch_geometric#5589] [General]
Hi.I am a new user of Graph neural Network. While I was looking at the code on internet I got the error message on the exam code. But I could not find where I make a mistake. How can I do this ?  The error is  invalid syntax for (conv1): GATConv(2, 32, heads=2)

##############################################

odel = GCN(hidden_channels=16)
optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)
criterion = torch.nn.CrossEntropyLoss()

def train():
      model.train()
      optimizer.zero_grad()
      out = model(data.x, data.edge_index)
      loss = criterion(out[data.train_mask], data.y[data.train_mask])
      loss.backward()
      optimizer.step()
      return loss

def test():
      model.eval()
      out = model(data.x, data.edge_index)
      pred = out.argmax(dim=1)
      test_correct = pred[data.test_mask] == data.y[data.test_mask]
      test_acc = int(test_correct.sum()) / int(data.test_mask.sum())
      return test_acc

for epoch in range(1, 101):
    loss = train()
    print(f'Epoch: {epoch:03d}, Loss: {loss:.4f}')
GAT(
  (conv1): GATConv(1433, 8, heads=8)
  (conv2): GATConv(64, 7, heads=8)
)
####################################

**Comment by rusty1s** [comment]
Simply removing
```
GAT(
(conv1): GATConv(1433, 8, heads=8)
(conv2): GATConv(64, 7, heads=8)
)
```
from the code should resolve this. I assume this relates to console output and not meant to be part of your script.

**Reply by AhmetGG** [reply]
Thank you for your reply. I tried to extract that part before but I still get error but different. its error message is __init__() missing 2 required positional arguments: 'in_channels' and 'out_channels' for first line. where can I focus on for channel?

**Reply by rusty1s** [reply]
Try initializing the model via `in_channels=dataset.num_features` and `out_channels=dataset.num_classes`.
